cinema/views.py

```diff
@@ -45,9 +72,39 @@ def get_serializer_class(self):
 
 
 class MovieSessionViewSet(viewsets.ModelViewSet):
-    queryset = MovieSession.objects.all()
+    queryset = MovieSession.objects.all().select_related(
+        "movie", "cinema_hall"
+    )
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+        date_param = self.request.query_params.get("date")
+        movie_id_param = self.request.query_params.get("movie")
+
+        if self.action == "list":
+            queryset = queryset.annotate(
+                tickets_available=F(
```

Comment:
what about queryset optimization ? select/prefect related ?

Reply:
But I optimized it before. here:
    queryset = MovieSession.objects.all().select_related(
        "movie", "cinema_hall"
    )
    so I don't have additional queries:
    
![image](https://github.com/mate-academy/py-tickets-orders/assets/37306964/807a3a44-4a06-460b-8a23-457bbd50ee34)
![image](https://github.com/mate-academy/py-tickets-orders/assets/37306964/cf88630c-4252-4f9b-90ae-55d9be995900)